### ISSUE TITLE
Multi-output kernel naming & consistency

### ISSUE THREAD

**Issue by st--** [JuliaGaussianProcesses/KernelFunctions.jl#328]
Currently there are several different multi-output kernels:

IndependentMOKernel, LatentFactorMOKernel, IntrinsicCoregionMOKernel, LinearMixingModelKernel.

There was [some debate](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/310#discussion_r661490968) about renaming
```diff
LinearMixingModelKernel
LinearMixingMOKernel
```
because
- A kernel is not a model so we should drop the `Model`.
- Currently the majority of MO kernels have the `MO` in the name, and we should be consistent.

The discussion did not belong on that PR, so teased out into its own issue here.

**Comment by willtebbutt** [comment]
I'm pro- `LinearMixingMOKernel` for the reasons that you state.

**Comment by devmotion** [comment]
I am for removing the `MO` from all kernels and the `Model` from `LinearMixingModelKernel` for consistency.

**Comment by willtebbutt** [comment]
One issue with this @devmotion is that things like `IndependentMOKernel` would become `IndependentKernel`, which is quite ambiguous.

**Comment by devmotion** [comment]
My impression was that we don't really want `IndependentMOKernel` in its current form (https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/issues/325), so we might want to choose a different more descriptive name anyway. Apart from that, I think the naming would still be consistent if it is called e.g. `IndependentMultiOutputKernel` and `MO` is dropped from all other kernels that clearly are kernels for multiple outputs and where there is no need to disambiguate them from a corresponding single output version.

**Comment by willtebbutt** [comment]
Fair enough. I'm very happy to bow to whatever the consensus is on this as I don't have particularly strong feelings either way.

**Comment by 4aHxKzD** [comment]
I don't like abbreviations used in programming so I'm in favor of removing the `MO` from the concrete types' names and using `MultiOutput` whenever there is ambiguity.

Is it too much to ask for renaming `ColVecs` to `ColumnVectors` and `RowVecs` to `RowVectors`? :pleading_face: :sweat_smile:

**Comment by willtebbutt** [comment]
>  Is it too much to ask for renaming ColVecs to ColumnVectors and RowVecs to RowVectors? 🥺 😅

Not at all! I would be very much in favour of that.

**Comment by st--** [comment]
I'd be happy with spelling out IndependentMultiOutputKernel but why would you keep it there but not for e.g. LinearMixingMultiOutputKernel?
